src/networkshared.h

```diff
@@ -559,6 +559,9 @@ class RingBuffer
 	}
 	void clear ( )
 	{
+		if ( _data != NULL )
+			delete[] _data;
```

Comment:
You'll also need to call `clear()` in the destructor of `RingBuffer` to free the memory when the instance is destroyed when it falls out of scope.

Reply:
I added another commit which adds a destructor inside `RingBuffer` to call `clear()`.

Reply:
Thanks!